### PR TITLE
fix(cosh-ng): [shell] reject zero idle timeout

### LINKED ISSUE #2094: [TestAgent][cosh-ng] bug: COSH_SHELL_EVIDENCE_IDLE_TIMEOUT_SECS=0 时 stalled fallback 风暴
### Component

cosh-ng

### Bug Description

`shell_evidence_idle_timeout` 直接 `value.parse::<u64>()` 解析 `COSH_SHELL_EVIDENCE_IDLE_TIMEOUT_SECS` 环境变量,无下界校验。当用户(或 systemd unit EnvironmentFile)误设为 `0` 时,`Duration::from_secs(0)` 让 `active_run_has_stalled_shell_evidence_delivery` 永远返回 true,`should_start_stalled_provider_shell_fallback` 在每个 poll 周期都触发 stalled fallback,形成 fallback 风暴,主循环空转、CPU 占满、日志暴增。

### Steps to Reproduce

```shell
ssh root@<ECS-IP>
# 静态确认:看 timeout 解析
sed -n '465,476p' src/cosh-ng/crates/cosh-shell/src/agent/poll.rs
# 看 stalled 判定
sed -n '487,495p' src/cosh-ng/crates/cosh-shell/src/agent/poll.rs
```

> 注:本 bug 为静态源码确认,动态复现需 `agent::poll` 模块编译进 test 二进制(本仓 build 配置下 `public.rs` 未加载 `mod.rs`,见测试报告关键发现 6)。

### Actual Behavior

`src/cosh-ng/crates/cosh-shell/src/agent/poll.rs:469-476`:

```rust
fn shell_evidence_idle_timeout() -> Duration {
    Duration::from_secs(
        std::env::var("COSH_SHELL_EVIDENCE_IDLE_TIMEOUT_SECS")
            .ok()
            .and_then(|value| value.parse::<u64>().ok())
            .unwrap_or(DEFAULT_SHELL_EVIDENCE_IDLE_TIMEOUT_SECS),
    )
}
```

`value.parse::<u64>()` 接受 `0`,`Duration::from_secs(0)` = 零超时。配合 `active_run_has_stalled_shell_evidence_delivery`(line 465-467):

```rust
fn active_run_has_stalled_shell_evidence_delivery(active_run: &ActiveAgentRun) -> bool {
    active_run.last_activity_at.elapsed() >= shell_evidence_idle_timeout()
}
```

`elapsed()` 永远 >= 0,函数永远返回 true;`should_start_stalled_provider_shell_fallback`(line 487-495)在每个 poll 周期都返回 true,fallback 风暴。

### Expected Behavior

`COSH_SHELL_EVIDENCE_IDLE_TIMEOUT_SECS=0` 应被拒(回退到 `DEFAULT_SHELL_EVIDENCE_IDLE_TIMEOUT_SECS`),或 parse 时设下界(如 `>= 1`)。

### Root Cause

`poll.rs:469-476` 的 `parse::<u64>` 无下界,接受 `0`。触发条件:用户/系统误设 `COSH_SHELL_EVIDENCE_IDLE_TIMEOUT_SECS=0`,或用 `systemd-edit-service` 设 `Environment=...=0`。设计取舍:无文档说明 0 应被允许;非 by-design,是数值配置缺下界校验。

### Suggested Fix

parse 后加 `.filter(|&v| v >= 1)`(`.min(1)` 或回退 default),或用 `parse::<u64>().ok().filter(|&v| v >= MIN_IDLE_TIMEOUT_SECS)`;同时加单测断言 0 触发 fallback → default 行为。

### Environment

- OS: Alibaba Alinux 4, kernel 6.6.102
- Component: cosh-ng
- Package/RPM: cosh-ng-0.13.0-2.alnx4

### Additional Context

- 相关 issue/PR: 无
- 稳定性: 静态源码确认(动态复现需 `agent::poll` 模块编译进 test 二进制)
- 被测版本 main HEAD: a96622bd

### PR DESCRIPTION
**Pull request by kongche-jbw**
## Why

`COSH_SHELL_EVIDENCE_IDLE_TIMEOUT_SECS` was parsed as a bare `u64`, so a configured `0` (a typo, or an `EnvironmentFile` entry) became a zero idle window. `active_run_has_stalled_shell_evidence_delivery` compares the run's idle time with `>=`, and that always holds against `Duration::ZERO`, so every poll cycle of every active run was classified as stalled and restarted the shell-evidence fallback continuation. The result is a fallback storm: the poll loop spins, CPU saturates, and the log volume explodes.

## What changed

`shell_evidence_idle_timeout` now rejects a configured `0` and falls back to the existing non-zero default (15s), the same path a malformed value already takes. Clamping to one second was considered and rejected: it would silently invent a window the operator never asked for and still fire fallbacks far more aggressively than intended. Valid positive values, malformed values, and the missing-variable case behave exactly as before.

The env read is split from the decision into a pure `shell_evidence_idle_timeout_from_config(Option<&str>)` helper, so the zero / positive / malformed / missing boundaries are unit-testable without mutating the process-global environment.

## Related issue

closes #2094

## User / Agent impact

Setting `COSH_SHELL_EVIDENCE_IDLE_TIMEOUT_SECS=0` no longer disables the idle window; it is ignored in favour of the 15s default. No other configured value changes behaviour.

## Risk and compatibility

- [x] Public CLI, API, configuration, or documented behavior changed
- [ ] Privileged or security-sensitive behavior changed
- [ ] Cross-component contract changed
- [ ] Migration or rollback guidance is needed

Low risk. The only behaviour change is for the previously pathological `0` setting, which had no legitimate use (it made every run look permanently stalled). The value was undocumented, so no documentation contract is broken.

## Validation

Run from `src/cosh-ng`:

- `cargo fmt --all -- --check` — clean
- `cargo clippy --workspace --all-targets -- -D warnings` — clean
- `cargo test -p cosh-shell --bin cosh-shell agent::poll::` — 15 passed (includes the two new boundary tests)
- `cargo test -p cosh-shell --lib` — 1211 passed
- `cargo test -p cosh-shell --test logic` — 9 passed
- `cargo test -p cosh-shell --test raw_cli provider_handoff` — 30 passed; `... raw_cli_streamed_tool_fallback` — 2 passed (both suites configure this env var)
- `crates/cosh-shell/scripts/check-layout.sh` — layout audit passed

Negative control: with the new lower-bound guard removed, both added tests fail (`0` resolves to a `0ns` window), confirming they cover the reported defect.

Edge cases asserted: `0`, `00`, `1`, `30`, `""`, `-1`, `abc`, `1.5`, `5s`, `" 5"`, and an unset variable.

One pre-existing failure is unrelated to this change: `activity::runtime_tests::activity_tool_output_details_show_internal_output_ref_in_debug` fails identically on the unmodified base commit when the temporary directory path is long enough to wrap inside the rendered card.

## Documentation and rollback

No documentation changes — the variable is not documented, and the fix only rejects a value that could never work. Rollback is a single-commit revert; there is no state or migration involved.

### CODE REVIEW

File: src/cosh-ng/crates/cosh-shell/src/agent/poll.rs [hunk 1]
```diff
@@ -538,10 +538,25 @@ fn active_run_has_stalled_shell_evidence_delivery(active_run: &ActiveAgentRun) -
...
+}
+
+// Resolve the idle window from its configured value, falling back to the
+// default when it is missing, malformed, or zero. Zero has to be rejected
+// rather than honoured: `elapsed() >= Duration::ZERO` holds on the very first
+// poll of a run, so a zero window would report every active run as stalled and
+// restart the fallback continuation on every poll cycle. Parsing is split out
+// of the env read so the boundaries stay testable without mutating the
+// process-global environment.
+fn shell_evidence_idle_timeout_from_config(configured: Option<&str>) -> Duration {
```
[the remark this answers is not in the document]
Comment (kongche-jbw) [reply src/cosh-ng/crates/cosh-shell/src/agent/poll.rs:555]: Not a real issue：负数会在 parse::<u64>() 阶段被拒绝，并与其他 malformed 值一样回退 default；helper 注释已明确 malformed 的行为，测试也覆盖了 "-1"。因此无需额外修改。

File: src/cosh-ng/crates/cosh-shell/src/agent/poll/tests.rs [hunk 2]
```diff
@@ -252,6 +252,45 @@ fn stalled_shell_evidence_delivery_uses_last_activity_idle_time() {
...
+        shell_evidence_idle_timeout_from_config(Some("1")),
+        Duration::from_secs(1)
+    );
+
+    let default = Duration::from_secs(DEFAULT_SHELL_EVIDENCE_IDLE_TIMEOUT_SECS);
+    // #2094: a configured `0` must fall back like a malformed value instead of
+    // becoming a zero idle window.
+    assert_eq!(shell_evidence_idle_timeout_from_config(Some("0")), default);
+    assert_eq!(shell_evidence_idle_timeout_from_config(Some("00")), default);
+    for malformed in ["", "-1", "abc", "1.5", "5s", " 5"] {
```
[the remark this answers is not in the document]
Comment (kongche-jbw) [reply src/cosh-ng/crates/cosh-shell/src/agent/poll/tests.rs:271]: Not a real issue：严格解析是本 PR 前的既有行为，旧实现同样不会接受带前导空格的 " 5"。在这里 trim 会扩大 #2094 之外的配置语义；该用例用于固定除零值外行为不变，因此无需修改。
